jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/component/NamedEnvironment.java

```diff
@@ -0,0 +1,54 @@
+package org.eclipse.jetty.util.component;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+import org.eclipse.jetty.util.Attributes;
+import org.eclipse.jetty.util.TypeUtil;
+
+class NamedEnvironment extends Attributes.Mapped implements Environment, Dumpable
```

Comment:
I don't really like creating a new top level class for such a small trivial thing.  What was wrong with Environment.Named as a package or private inner class?

Reply:
The class was implicitly public with no way to alter its visibility as it was enclosed by an interface.

Maybe `Environment` should be a class?

Reply:
hmmm or just moved NamedEnvironment to an internal sub package